From fdc3df03dd57b0348faa2c932661902ba34fbdc9 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Caio=20Rom=C3=A3o?= <caioromao@gmail.com>
Date: Sun, 14 Jun 2009 16:55:41 -0300
Subject: [PATCH 02/10] bugzilla.py: Support for custom filename.

attach() now has a new keyword parameter called `filename_override`
which, when not None, replaces the name given by the `filename`
parameter.
---
 bugz/bugzilla.py |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/bugz/bugzilla.py b/bugz/bugzilla.py
index df9e13c..d2b1e97 100644
--- a/bugz/bugzilla.py
+++ b/bugz/bugzilla.py
@@ -758,7 +758,7 @@ class Bugz:
 		return 0
 
 	def attach(self, bugid, title, description, filename,
-			content_type = 'text/plain'):
+			content_type = 'text/plain', filename_override=None):
 		"""Attach a file to a bug.
 
 		@param bugid: bug id
@@ -771,6 +771,8 @@ class Bugz:
 		@type  filename: string
 		@keywords content_type: mime-type of the attachment
 		@type content_type: string
+        @keywords filename_override: name to be used instead of filename
+        @type filename_override: string
 
 		@rtype: bool
 		@return: True if successful, False if not successful.
@@ -784,7 +786,8 @@ class Bugz:
 		qparams['comment'] = description
 		qparams['contenttypeentry'] = content_type
 
-		filedata = [('data', filename, open(filename).read())]
+		filedata = [('data', filename_override or filename,
+                     open(filename).read())]
 		content_type, body = encode_multipart_formdata(qparams.items(),
 				filedata)
 
-- 
1.6.3.2

